db/comdb2.h

```diff
@@ -3325,8 +3336,9 @@ int read_spfile(char *file);
 struct bdb_cursor_ifn;
 int recover_deadlock(bdb_state_type *, struct sql_thread *,
                      struct bdb_cursor_ifn *, int sleepms);
```

Comment:
From review: make this recover_deadlock_flags_int, check the rcode in recover_deadlock_flags, and do all cleanup (put_curtran, close_cursors, assert lockref == 0) if rcode is non-zero.